enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/state/CoreState.java

```diff
@@ -121,7 +121,7 @@ private void submitApplyJob( long lastToApply )
             {
                 lastApplyingStorage.persistStoreData( lastToApply );
 
-                while ( cursor.next() && cursor.index() <= lastToApply )
+                while ( cursor.index() < lastToApply && cursor.next() )
```

Comment:
It isn't really an issue that can be observed easily from the outside, making it hard to write a unit test.